src/System.Private.Xml.Linq/tests/TreeManipulation/SaveWithFileName.cs

```diff
@@ -0,0 +1,249 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using CoreXml.Test.XLinq;
+using Microsoft.Test.ModuleCore;
+using XmlCoreTest.Common;
+using Xunit;
+
+namespace XLinqTests
+{
+    public class SaveWithFileName
+    {
+        private const string BaseSaveFileName = "baseSave.xml";
+        private const string TestSaveFileName = "testSave";
+        private const string xml = "<PurchaseOrder><Item price=\"100\">Motor<![CDATA[cdata]]><elem>innertext</elem>text<?pi pi pi?></Item></PurchaseOrder>";
+        private const string xmlForXElementWriteContent = "<Item price=\"100\">Motor<![CDATA[cdata]]><elem>innertext</elem>text<?pi pi pi?></Item>";
+
+        //[Variation(Priority = 0, Desc = "Save and validate XDocument")]
+        [Fact]
+        public void writer_1()
+        {
+            XDocument document = new XDocument(new XElement("PurchaseOrder", new XElement("Item", "Motor", new XAttribute("price", "100"),
+                new XCData("cdata"), new XElement("elem", "innertext"), new XText("text"), new XProcessingInstruction("pi", "pi pi"))));
+            SaveBaseline(xml);
+
+            SaveWithFile(document);
```

Comment:
I can't find where these first 3 tests actually assert anything. Can you? I looked at the code of FilePathUtil.